Sources/Either/Either.swift

```diff
@@ -0,0 +1,155 @@
+public enum Either<Left, Right> {
+  case left(Left)
+  case right(Right)
+
+  public var left: Left? {
+    return self.either(ifLeft: Optional.some, ifRight: { _ in Optional.none })
+  }
+
+  public var right: Right? {
+    return self.either(ifLeft: { _ in Optional.none }, ifRight: Optional.some)
+  }
+
+  public func either<Value>(
+    ifLeft: (Left) throws -> Value,
+    ifRight: (Right) throws -> Value
+    ) rethrows -> Value {
+    switch self {
+    case let .left(left):
+      return try ifLeft(left)
+    case let .right(right):
+      return try ifRight(right)
+    }
+  }
+
+  public func `do`(
+    ifLeft: (Left) throws -> Void,
+    ifRight: (Right) throws -> Void
+    ) rethrows {
+    switch self {
+    case let .left(left):
+      try ifLeft(left)
+    case let .right(right):
+      try ifRight(right)
+    }
+  }
+
+  public func bimap<NewLeft, NewRight>(
+    ifLeft transformLeft: (Left) throws -> NewLeft,
+    ifRight transformRight: (Right) throws -> NewRight
+    ) rethrows -> Either<NewLeft, NewRight> {
+    return try self.either(
+      ifLeft: { .left(try transformLeft($0)) },
+      ifRight: { .right(try transformRight($0)) }
+    )
+  }
+
+  public func mapLeft<NewLeft>(
```

Comment:
Added unbiased `mapLeft`, `mapRight`, `flatMapLeft`, `flatMapRight`, etc. Do we want to add `zipLeft` and `zipRight` too?

Reply:
`zipLeft/Right` makes sense to me!